samples/net/lwm2m_client/src/lwm2m-client.c

```diff
@@ -11,6 +11,7 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
+#include <hwinfo.h>
```

Comment:
This should probably be wrapped with #if defined(CONFIG_HWINFO)

Reply:
I my tests this header is always available even without CONFIG_HWINFO enabled.

Can't wait to see your improvements. ;)